spec/index.bs

```diff
@@ -596,24 +596,36 @@ Issue(gpuweb/gpuweb#296): Consider merging all read-only usages.
 Textures may consist of separate [=mipmap levels=] and [=array layers=],
 which can be used differently at any given time.
 Each such <dfn dfn>subresource</dfn> is uniquely identified by a
-[=texture=], [=mipmap level=], and
-(for {{GPUTextureDimension/2d}} textures only) [=array layer=].
+[=texture=], [=mipmap level=],
+(for {{GPUTextureDimension/2d}} textures only) [=array layer=],
+and [=aspect=].
 
-The **main usage rule** is that any [=subresource=]
-at any given time can only be in either:
+<dfn dfn>atom resource</dfn> means a whole buffer or a single subresource of a texture.
+
+The **main usage rule** is that any [=atom resource=]
+at any given time within [=usage scope=] can only be in either:
   - a combination of [=read-only usage=]s
   - a single [=mutating usage=]
 
+The only exception is that a combination of writeonly-storage-texture usages, or a combination
+of storage-buffer usages upon the same [#atom resource#] in render pass is allowed.
+
+Note that all invisible usages and overwritten usages should be counted, for example:
+  - a resource binding attached to a shader stage with visiblity none
+  - a resource binding attached to compute shader stage in render pass, or vice versa
+  - a collection of bindings are set by a bind group but being overwritten within current [=usage scope=]
```

Comment:
Can this be reworded to avoid saying "overwritten" (since we don't define what this means)? We could say something like "resources bound in a group that is replaced by a different bind group before any draws or dispatches are recorded"

Reply:
Done. Good suggestion!